apps/openmw/mwphysics/projectile.cpp

```diff
@@ -0,0 +1,80 @@
+#include "projectile.hpp"
+
+#include <BulletCollision/CollisionShapes/btSphereShape.h>
+#include <BulletCollision/CollisionDispatch/btCollisionWorld.h>
+
+#include <components/sceneutil/positionattitudetransform.hpp>
+#include <components/resource/bulletshape.hpp>
+#include <components/debug/debuglog.hpp>
+#include <components/misc/convert.hpp>
+
+#include "../mwworld/class.hpp"
+
+#include "collisiontype.hpp"
+
+namespace MWPhysics
+{
+
+
+Projectile::Projectile(int projectileId, const osg::Vec3f& position, btCollisionWorld* world, bool isMagicBolt)
+    : mCollisionWorld(world)
+{
+    mProjectileId = projectileId;
+
+    float radius = isMagicBolt ? 20.f : 5.f;
+    mShape.reset(new btSphereShape(radius));
+    mHalfExtents = osg::Vec3f(radius, radius, radius);
+    mMeshTranslation = osg::Vec3f(0.f, 0.f, 0.f);
+    mConvexShape = static_cast<btConvexShape*>(mShape.get());
+
+    mCollisionObject.reset(new btCollisionObject);
+    mCollisionObject->setCollisionFlags(btCollisionObject::CF_KINEMATIC_OBJECT);
+    mCollisionObject->setActivationState(DISABLE_DEACTIVATION);
+    mCollisionObject->setCollisionShape(mShape.get());
+    mCollisionObject->setUserPointer(this);
+
+    setPosition(position);
+
+    addCollisionMask(getCollisionMask());
```

Comment:
The collision mask is constant (it can even be constexpr), so addCollisionMask and getCollisionMask can be removed in favor of only adding the collision object in the constructor. I don't suppose anything needs to work with a projectile's collision mask.